mapper/orm/src/main/java/org/hibernate/search/mapper/orm/automaticindexing/session/impl/ConfiguredAutomaticIndexingSynchronizationStrategy.java

```diff
@@ -45,7 +46,8 @@ public DocumentRefreshStrategy getDocumentRefreshStrategy() {
 
 	public void executeAndSynchronize(PojoIndexingPlan indexingPlan) {
 		CompletableFuture<SearchIndexingPlanExecutionReport> reportFuture =
-				indexingPlan.executeAndReport( entityReferenceFactory ).thenApply( SearchIndexingPlanExecutionReportImpl::new );
+				indexingPlan.executeAndReport( entityReferenceFactory, OperationSubmitterType.BLOCKING )
```

Comment:
I believe the submitter should be configurable in `AutomaticIndexingSynchronizationConfigurationContext`, as we discussed on Zulip.